pyls/server.py

```diff
@@ -32,6 +43,9 @@ def handle(self):
                 log.exception("Language server shutting down for uncaught exception")
                 break
 
+            if isinstance(self.wfile, socket_type):
+                done = True
```

Comment:
Why is done true for a socket? I think you're trying to make this support HTTP instead of the long-running TCP server. 

Check out the vscode-client project and modify src/extension.ts to use the TCP port. You will see that VSCode functions correctly sending frames over TCP